src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/AesGcm.macOS.cs

```diff
@@ -0,0 +1,128 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Security.Cryptography
+{
+    public sealed partial class AesGcm
+    {
+        // Apple CryptoKit does not support short authentication tags. Since .NET originally supported AES-GCM via
+        // OpenSSL, which does support short tags, we need to continue to support them. If a caller supplies a short
+        // tag we will continue to use OpenSSL if it is available. Otherwise, use CryptoKit.
+        private static readonly bool s_openSslAvailable = Interop.OpenSslNoInit.OpenSslIsAvailable;
+        private const int CryptoKitSupportedTagSizeInBytes = 16;
+
+        private byte[]? _key;
+
+        // CryptoKit added AES.GCM in macOS 10.15, which is our minimum target for macOS. We still may end
+        // up throwing a platform not supported if a caller uses a short authentication tag and OpenSSL is not
+        // available. But recommended use of AES-GCM with a 16-byte tag is supported.
+        public static bool IsSupported => true;
+
+        [MemberNotNull(nameof(_key))]
+        private void ImportKey(ReadOnlySpan<byte> key)
+        {
+            // We should only be calling this in the constructor, so there shouldn't be a previous key.
+            Debug.Assert(_key is null);
+
+            // Pin the array on the POH so that the GC doesn't move it around to allow zeroing to be more effective.
+            _key = GC.AllocateArray<byte>(key.Length, pinned: true);
+            key.CopyTo(_key);
+        }
+
+        private void EncryptCore(
+            ReadOnlySpan<byte> nonce,
+            ReadOnlySpan<byte> plaintext,
+            Span<byte> ciphertext,
+            Span<byte> tag,
+            ReadOnlySpan<byte> associatedData)
+        {
+            CheckDisposed();
+
+            if (tag.Length != CryptoKitSupportedTagSizeInBytes)
+            {
+                using (SafeEvpCipherCtxHandle ctxHandle = CreateOpenSslHandle())
+                {
+                    AesGcmOpenSslCommon.Encrypt(ctxHandle, nonce, plaintext, ciphertext, tag, associatedData);
+                }
+            }
```

Comment:
Aren't GCM tags just truncated?  Therefore, if we care to keep support for various tag sizes, this would just need to stackalloc a big enough buffer to make CryptoKit happy and copy the answer over to `tag` when done... no OpenSSL interaction required.

Decrypt would be where it gets trickier.

Reply:
> Decrypt would be where it gets trickier.

Right. It would be a little weird for encrypt to work and decrypt to not, so I kinda wanted symmetry there.